modules/internal/String.chpl

```diff
@@ -217,26 +220,70 @@ module String {
     var shortData : chpl__inPlaceBuffer;
   }
 
+  /*
+     A value of type :record:`byteIndex` can be passed to certain
+     `string` functions to indicate that the function should operate
+     with units of bytes. See :proc:`~string.this`.
+
+     To create or modify a :record:`byteIndex`, cast it to or from an
+     `int`. For example, the following function returns a string
+     containing only the second byte of the argument:
```

Comment:
I think this would be a good place to mention the arithmetic operators supported.